src/include/txn_inline.h

```diff
@@ -842,11 +849,12 @@ __wt_txn_pinned_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *pinned_tsp)
      * pinned. If a checkpoint is starting and we have to use the checkpoint timestamp, we take the
      * minimum of it with the oldest timestamp, which is what we want.
      */
-    WT_ACQUIRE_BARRIER();
     checkpoint_ts = txn_global->checkpoint_timestamp;
```

Comment:
@ajmorton  Not sure why there is no warnings for this read. checkpoint_timestamp is a shared variable and should be read using atomic read.

Reply:
Annoyingly there's no way to check this. We could if we moved to _Atomic but MSVC support is still lacking. For now we're relying on developers to use the shared vars correctly.

Reply:
Do you want me to fix this in the pr?

Reply:
No that's ok. txn_global is going to need a much larger rework in a future project. I don't think we need to add noise to this PR.

Reply:
Thanks